scripts/apps/monitoring/helpers/CustomSortOfGroups.ts

```diff
@@ -1,8 +1,19 @@
 import {get} from 'lodash';
 import {StageGroup} from '../directives/MonitoringGroup';
 
+type OrderType = 'desc' | 'asc';
+
 export type GroupSortOptions = Array<string>;
 
+export type GroupSortConfig = {
+    default?: string, // field:order (publish_schedule:asc)
```

Comment:
I had that before, but the config in `superdesk.config.js` was getting more complex, because I could put `default: { field: '', order: '' }`. And if we don't put the order (optional) then the object would be pointless. I could also keep the current config  and use the object in the code, but I wanted to keep GroupSortConfig the same for config and code

Reply:
>  was getting more complex, because I could put default: { field: '', order: '' }. And if we don't put the order (optional) then the object would be pointless

I'm not sure I understand what you mean. If you want a config a default, you set `{default: {field: 'headline', order: 'asc'}}` and if you don't, you don't define it or set to null. There's no need to have an empty dummy object.

Reply:
I meant the `order` field is optional

Reply:
Why not make it non-optional? :wink: If someone wants a custom sorting, let's have them define the order too.

Reply:
I don't see how it's relevant if the order is optional or not. If the object is `null`, you can apply the default one if you want. How is this related to having a config option as a string combined from multiple fields?

Reply:
Yeah you might be right. I'm still not sure if I should make it an object, the current config is complex enough

Reply:
```
monitoring: {
    scheduled: {
        sort: {
            default: {
                field: 'publish_schedule'
                order: 'asc'
            },
            options: [ 'publish_schedule' ]
        }
    },
}
```

I don't know, it doesn't seem complex to me. What doesn't seem clear enough for me is whenever `scheduled` can be any stage name and `options` is too vague, I'd use 'allowed_article_fields'. Other than that, it looks pretty straight-forward to me.

Reply:
Maybe `article_fields_allowed_to_sort_by` is too long and as it's already inside a `sort` object, I can just name it `article_fields` or something like that.

Reply:
I want names to first be clear and then short if possible. `sort: {article_fields: []}` might as well mean `sortBy(article_fields[0]).thenBy(article_fields[1]).thenBy(article_fields[2])`.